tests/tracee_test.go

```diff
@@ -126,3 +129,92 @@ func TestWebhookIntegration(t *testing.T) {
 	t.Log("Terminating the Tracee container...")
 	assert.NoError(t, exec.Command("docker", "kill", containerID).Run())
 }
+
+type traceeContainer struct {
+	testcontainers.Container
+}
+
+func setupTraceeContainer(ctx context.Context, tempDir string) (*traceeContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:      "tracee",
+		Privileged: true,
+		BindMounts: map[string]string{tempDir: tempDir},
+		Name:       "tracee",
+		AutoRemove: true,
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &traceeContainer{Container: container}, nil
+}
+
+func setupTraceeTrainerContainer(ctx context.Context, sigid string) (*traceeContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:      "tracee-trainer",
+		Entrypoint: []string{"/runner", sigid},
+		Privileged: true,
+		Name:       "tracee-trainer",
+		AutoRemove: true,
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &traceeContainer{Container: container}, nil
+}
+
+func TestTraceeSignatures(t *testing.T) {
+	tempDir := "/tmp/tracee"
+	err := os.MkdirAll(tempDir, os.ModePerm) // TODO: Investiagte why passing ioutil.Temp does not work with Tracee mount
+	require.NoError(t, err)
+	defer func() {
+		os.RemoveAll(tempDir)
+	}()
+
+	for _, sigid := range []string{"TRC-3", "TRC-4", "TRC-9", "TRC-10", "TRC-11"} {
+		t.Run(sigid, func(t *testing.T) {
+			ctx := context.Background()
+
+			// run tracee container
+			traceeContainer, err := setupTraceeContainer(ctx, tempDir)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer traceeContainer.Terminate(ctx)
+
+			// wait for tracee to initialize
+			time.Sleep(time.Second * 5)
```

Comment:
Ideally, we should be using ContainerRequest.WaitStrategy at the time when we define the test container:

```go
req := testcontainers.ContainerRequest{
	Image:      "tracee",
	Privileged: true,
	BindMounts: map[string]string{tempDir: tempDir},
	Name:       "tracee",
	AutoRemove: true,
	WaitingFor: wait.ForLog("Loaded 5 signature(s)"),
}
```

Reply:
Yes you're correct. I just found out about this and will implement.